### ISSUE TITLE
Tab rollover highlighting/Divider lines/Custom Themes/Classic Theme Restorer

### ISSUE THREAD

**Issue by Mr-Personality** [Croydon/vertical-tabs-reloaded#24]
First of all, what on earth is this place? "Git hub"? What… a forum full of gits? "Hello world"? "Repository"? "Pull requests"? It's all very strange. But anyway, here's a bit of feedback about Vertical Tabs Reloaded…

Great add-on! This seems like the vertical tabs addon everyone has wanted for years but didn't exist. Until lately I had been using Tab Kit 2nd Edition, but it broke in Firefox v45.

So well done Croydon and thank you! :)

Here are four humble suggestions for improvement…

- [x] **1. Addon Clash**

When I first installed it, both the tab background color and text were "all white" so I couldn't see them. I quickly discovered it's because I use the Classic Theme Restorer addon (there's no way I would ever uninstall this as it disables the horrible new Firefox search bar).

I was able to solve the "all white" problem by changing Classic Theme Restorer's tab style to "classic tabs" which then allows your addon to override the colors. I wanted you to be aware of this, and was wondering if it's possible for your addon to detect if Classic Theme Restorer is installed? If it does detect Classic Theme Restorer, it should set the style to "Windows" by default, thus preventing the "all white" problem.

- [x] **2. Lack of rollover highlighting**

I don't know whether it's a bug, or deliberate, but when I use "Darwin" or "Linux", the tabs don't change color when I roll over them. To me, this makes "Darwin" and "Linux" both completely useless themes, since I definitely want tabs to change color when I roll over them. The "Windows" theme is little better and only has an outline around the tabs when you roll over. I'm sure I can't be the only one who doesn't like this, and prefers a solid rollover color so you can clearly see which tab you're selecting.

- [x]  **3. No divider lines**

With the exception of "Light", none of your styles have divider lines between the tabs. For me, divider lines are really important. Without them, the tabs don't look distinct enough and just kind of blur into one another and look untidy. It's harder to read them somehow. So I am using the Light style, which has dividers, though I find the tab height of Light a little too tall. Ideally I would like a style like "Light", with the height of "Linux". Or better still, the ability to change the tab height.

- [ ] **4. Custom colors**

I don't know how hard it would be to implement, but it would be great to have custom colors, or to be able to submit colors for inclusion as a new style. My color choice would be something like this:

Unselected tab background = c9c9c9
Unselected tab text = 5c5c5c
Selected/rollover tab background = f7f7f7
Selected/rollover tab text = 0
Divider = a4a4a4

Thanks :)

**Comment by Croydon** [comment]
@Mr-Personality Thank you for your feedback and welcome on GitHub!

GitHub is a place to develop together easily (open source) software. Issues are there to discuss, organise and plan what there is to do. Pull requests are changes somebody made to the code and would like to see them in the original project. If you would like to know more about GitHub itself I am pretty sure you will find enough resources for your questions. 

Also please open in the future for every single suggestion an own issue or there is a good chance, that it will get a mess. Thank you. 

#### 1. Add-on compatibility
I am planing to offer some kind of vanilla/plain theme, so that other add-ons can fully change the colors, background and so one of the tabs. Of course it will never work 100% since VTR need to manipulate the tabs so far, that there are visible, functional and vertical. 

#### 2. Hover highlighting
Thanks for reporting this. The Linux and Darwin theme will have a hover/rollover color starting with v0.4.1.

#### 3. No divider lines
Besides the Light theme, the Darwin theme already had a border between tabs as well. Starting with v0.4.1 all themes will have borders between tabs. You are absolutely right, that it is easier to orientate with borders.

#### 4. Custom colors
I am planing to change the way how themes are working internally. Before I made this changes I don't want to make major changes to the existing themes nor do it want to add new ones. Afterwards it will be a lot easier to do so. I'm not sure if I will add at some point the option to fully customize the colors individually via the settings, but I am considering this. :)

**Comment by Mr-Personality** [comment]
That's great! Thank you Croydon. Good work! :)

**Comment by Mr-Personality** [comment]
That's great! Thank you Croydon. Good work! :) Am I supposed to "close issue" now you have answered my questions?

**Comment by Croydon** [comment]
@Mr-Personality You should only close issues on your own when either it turned out to be invalid (e.g. you report a bug which isn't one) or if it is a duplicate of an already existing issue. When this doesn't match or in doubt it is better to leave the decision to the project maintainer(s). I will close this issue if everything you suggested is either implemented or if I decided against it.

**Comment by Mr-Personality** [comment]
Great, thanks :)

**Comment by Croydon** [comment]
@Mr-Personality I the meantime I have added the possibility to select no theme at all, it still injects a few style rules to make the tab sidebar usable, but it concentrates on the absolutely necessary stuff.
Therefore the no theme option could be the best choice to make sure that it works together with other add-ons, which are manipulate the tabs.

I'm not ruling it out, that I might add in the future auto detection of other add-ons and set the preferences respectively, but for the moment I have decided against it.
